robot/impl/resource_manager_test.go

```diff
@@ -735,9 +735,6 @@ func TestManagerNewComponent(t *testing.T) {
 		return resource.Name{}, false
 	}), test.ShouldBeNil)
 	robotForRemote.config.Components[8].DependsOn = append(robotForRemote.config.Components[8].DependsOn, "arm3")
-	_, err = config.SortComponents(robotForRemote.config.Components)
-	test.That(t, err, test.ShouldNotBeNil)
-	test.That(t, err.Error(), test.ShouldEqual, "circular dependency detected in component list between arm3, board3")
```

Comment:
we will still return a circular dependency error when we try to update the robot with this config, and this behavior is tested at the end of this test.